src/plugins/vis_type_vega/public/vega_view/vega_base_view.js

```diff
@@ -262,7 +267,13 @@ export class VegaBaseView {
         this._addDestroyHandler(() => tthandler.hideTooltip());
       }
 
-      return view.runAsync(); // Allows callers to await rendering
+      const state = this._vegaStateRestorer.restore();
```

Comment:
I think that we should not only restore signals after `kibanaAddFilter` or other vega expression function has been executed. From my perspective it make sense to restore it also on getting new data.

Reply:
Yes, this is a good change.